kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoKVTable.java

```diff
@@ -210,4 +207,11 @@ public long approximateNumEntries(final int kafkaPartition) {
     return 0;
   }
 
+  private ObjectId idForKafkaPartition(final int kafkaPartition) {
+    // allocate 16 bytes total since ObjectId needs 12 for internal stuff
+    byte[] bytes = new byte[16];
+    final ByteBuffer buffer = ByteBuffer.wrap(bytes).putInt(kafkaPartition);
```

Comment:
looks like there's some kind of bug here, it always creates the following ID when I'm running it in my test:
```
{"_id":{"$oid":"000000000000000000000000"},"partition":{"$numberInt":"2"},"epoch":{"$numberInt":"1"},"offset":{"$numberLong":"-1"}}
```
I'm going to go ahead and patch this since it's blocking some of my experiments.

Reply:
https://github.com/responsivedev/responsive-pub/pull/210